assembly/__tests__/vm-mock.spec.ts

```diff
@@ -0,0 +1,94 @@
+import {
+  Address,
+  Storage,
+  generateEvent,
+  Args,
+  toBytes,
+  Context,
+  fromBytes,
+} from "../std";
+
+const testAddress = new Address(
+  "A12E6N5BFAdC2wyiBV6VJjqkWhpz1kLVp2XpbRdSnL1mKjCWT6oR"
+);
+
+const keyTestArgs = new Args().add("test");
+const valueTestArgs = new Args().add("value");
+
+const keyTest = toBytes("test");
+const valueTest = toBytes("value");
+
+describe("Testing mocked Storage and CallStack", () => {
+  test("Testing the callstack", () => {
+    const callStack = Context.addressStack();
+    expect(callStack[0].toByteString()).toBe(
+      "A12UBnqTHDQALpocVBnkPNy7y5CndUJQTLutaVDDFgMJcq5kQiKq"
+    );
+    expect(callStack[1].toByteString()).toBe(
+      "A12BqZEQ6sByhRLyEuf0YbQmcF2PsDdkNNG1akBJu9XcjZA1eT"
+    );
+  });
+
+  test("Testing the Storage setOf", () => {
+    Storage.setOf(testAddress, keyTest, valueTest);
+    expect(
+      fromBytes(
+        Storage.getOf(
+          new Address("A12E6N5BFAdC2wyiBV6VJjqkWhpz1kLVp2XpbRdSnL1mKjCWT6oR"),
+          toBytes("test")
+        )
+      )
+    ).toBe("value");
```

Comment:
You should use string or bytes, here your mixing the two and then the test is less clear...